assignments/Session_02/http_server.py

```diff
@@ -0,0 +1,123 @@
+import socket
+import sys
+import os
+import mimetypes
+
+
+def response_ok(body=b"this is a pretty minimal response", mimetype=b"text/plain"):
+    """returns a basic HTTP response"""
+    resp = []
+    resp.append(b"HTTP/1.1 200 OK")
+    resp.append(b"Content-Type: " + mimetype)
+    resp.append(b"")
+    resp.append(body)
+    return b"\r\n".join(resp)
+
+
+def response_method_not_allowed():
+    """returns a 405 Method Not Allowed response"""
+    resp = []
+    resp.append("HTTP/1.1 405 Method Not Allowed")
+    resp.append("")
+    return "\r\n".join(resp).encode('utf8')
+
+
+def response_not_found():
+    """returns a 404 Not Found response"""
+    return b"HTTP/1.1 404 Not Found"
+
+
+def parse_request(request):
+    first_line = request.split("\r\n", 1)[0]
+    method, uri, protocol = first_line.split()
+    if method != "GET":
+        raise NotImplementedError("We only accept GET")
+    return uri
+
+
+def resolve_uri(uri):
+    """This method should return appropriate content and a mime type"""
+    home_directory = 'webroot'
+    relative_uri = home_directory + uri
+
+    """ What am I?  Several tests to determine whether the URI points to:
+    """
+    contents = "not yet filled"
+
+    # test to determine it the URI is a directory
+    #######
+    try:
+        contents = os.listdir(relative_uri)
+        comma = ','
+        contents = comma.join(contents)
+        contents = contents.encode('utf-8')
+        mimetype = b'text/plain'
+    except:
+        pass
+
+    # uri is a file, determine the mimetype and read the file
+    #######
+    if contents == "not yet filled":
+        mimetype = mimetypes.guess_type(relative_uri)[0]
+        mimetype = mimetype.encode()
+#############################################################################
+#Note to Cris - the mimetypes function returns a text/plain mimetype for a .py file rather than the
+# text/x-python that the test was expecting
```

Comment:
this is odd, and may be a windows thing.  I am curious if any of your other fellow students have run into the issue.